sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -214,21 +238,33 @@ class Response
 
 public:
   Response(uint16_t statusCode, std::string const& reasonPhrase)
-      : Response(statusCode, reasonPhrase, http::BodyBuffer::null, http::BodyStream::null)
+      : Response(statusCode, reasonPhrase, std::vector<uint8_t>(), http::BodyStream::null)
   {
   }
 
+  // default constructor for bulding from base response
+  Response() : Response(0, "", std::vector<uint8_t>(), http::BodyStream::null) {}
+
   // Methods used to build HTTP response
+  void setVersion(uint16_t mayorVersion, uint16_t minorVersion)
```

Comment:
Because of what I said above, this function should not exist. Only VALID Response object should be ever exist and a valid response requires a status code, version, reason phrase, and body buffer/stream.